be/src/runtime/stream_load/stream_load_executor.cpp

```diff
@@ -188,7 +188,7 @@ Status StreamLoadExecutor::commit_txn(StreamLoadContext* ctx) {
     // Return if this transaction is committed successful; otherwise, we need try to
     // rollback this transaction
     Status status(result.status);
-    if (!status.ok()) {
+    if (!status.ok() && status.code() != TStatusCode::PUBLISH_TIMEOUT) {
```

Comment:
If you change this code, when FE return PUBLISH_TIMEOUT, here it will just return Status:OK() to the upper caller?
I think we should return PUBLISH_TIMEOUT directly, but set ctx->need_rollback = false to false